src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,141 @@
+package duke.task;
+
+import duke.DukeException;
+import duke.Ui;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class that encapsulates a list of tasks.
+ */
+public class TaskList {
+    private final List<Task> tasks;
+    private final Ui ui = new Ui();
+
+    /**
+     * Returns a new empty TaskList object.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Returns a new TaskList object with the given tasks.
+     * @param tasks
+     */
+    public TaskList(List<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Gets the size of the task list.
+     * @return The size of the task list.
+     */
+    public int getSize() {
+        return this.tasks.size();
+    }
+
+    public List<Task> getTasks() {
+        return this.tasks;
+    }
+
+    /**
+     * Gets the task at the specified index.
+     * @param taskNo Specified index.
+     * @return The task at the specified index.
+     */
+    public Task getTask(int taskNo) {
+        return this.tasks.get(taskNo - 1);
+    }
+
+    /**
+     * Turns the task list into a String.
+     * @return The String representation of the list to be saved to the hard disk.
+     */
+    public String saveList() {
+        StringBuilder lst = new StringBuilder();
+        for (int i = 1; i <= this.getSize(); i++) {
+            lst.append(i).append(". ").append(this.getTask(i).toString()).append("\n");
+        }
+        return lst.toString();
+    }
+
+    /**
+     * Marks a specified task as done.
+     * @param taskNo The number of the task to be marked as done.
+     */
+    public void taskDone(int taskNo) {
```

Comment:
Maybe you could change the name of this method to `markTask`, since `taskDone` is not in the form of a verb?